docs/reference/search/profile.asciidoc

```diff
@@ -1047,6 +1046,224 @@ per hit. In that case the true cost of `_source` phase is hidden in the
+that uses `dfs_query_then_fetch`:
+
+Let's first setup an index with multiple shards and index
+a pair of documents with different values on a `keyword` field.
+
+[source,console,id=profile_dfs]
+--------------------------------------------------
+PUT my-dfs-index
+{
+  "settings": {
+    "number_of_shards": 2, <1>
+    "number_of_replicas": 1
+  },
+  "mappings": {
+      "properties": {
+        "my-keyword": { "type": "keyword" }
+      }
+    }
+}
+
+POST my-dfs-index/_bulk?refresh=true
+{ "index" : { "_id" : "1" } }
+{ "my-keyword" : "a" }
+{ "index" : { "_id" : "2" } }
+{ "my-keyword" : "b" }
+--------------------------------------------------
+<1> The `my-dfs-index` is created with multiple shards.
+
+With an index setup, we can now profile the dfs phase of a
+search query. For this example we use a term query.
+
+[source,console]
+--------------------------------------------------
+GET /my-dfs-index/_search?search_type=dfs_query_then_fetch&pretty&size=0 <1>
+{
+  "profile": true, <2>
+  "query": {
+    "term": {
+      "my-keyword": {
+        "value": "a"
+      }
+    }
+  }
+}
+--------------------------------------------------
+// TEST[continued]
+<1> The `search_type` url parameter is set to `dfs_query_then_fetch` to
+ensure the dfs phase is run.
+<2> The `profile` parameter is set to `true`.
+
+In the response, we see a profile which includes a `dfs` section
+for each shard along with profile output for the rest of the search phases.
+One of the `dfs` sections for a shard looks like the following:
+
+[source,console-result]
+--------------------------------------------------
+"dfs" : {
+    "statistics" : {
+        "type" : "statistics",
+        "description" : "collect term statistics",
+        "time_in_nanos" : 236955,
+        "breakdown" : {
+            "term_statistics" : 4815,
+            "collection_statistics" : 27081,
+            "collection_statistics_count" : 1,
+            "create_weight" : 153278,
+            "term_statistics_count" : 1,
+            "rewrite_count" : 0,
+            "create_weight_count" : 1,
+            "rewrite" : 0
+        }
+    }
+}
+--------------------------------------------------
+// TESTRESPONSE[s/^/{\n"took": $body.took,\n"timed_out": $body.timed_out,\n"_shards": $body._shards,\n"hits": $body.hits,\n"profile": {\n"shards": [ "$body.$_path", {\n"id": "$body.$_path",\n/]
+// TESTRESPONSE[s/}$/}, "aggregations": [], "searches": $body.$_path}]}}/]
+// TESTRESPONSE[s/(\-)?[0-9]+/ $body.$_path/]
+
+In the `dfs.statistics` portion of this response we see a `time_in_nanos`
+which is the total time it took to collect term statistics for this
+shard along with a further breakdown of the individual parts.
+
+[[profiling-knn-search]]
+====== Profiling kNN Search
+
+A <<approximate-knn, k-nearest neighbor (kNN)>> search runs during
+the dfs phase.
+
+The following is an example of setting `profile` to `true` on a search
+that has a knn section:
```

Comment:
Tiny comment, we usually style the names of API parameters as literals ("`knn` section").

Reply:
Good catch. Fixed.